packages/tab/index.js

```diff
@@ -0,0 +1,192 @@
+      setAttr: (attr, value) => this.setState({[attr]: value}),
+
+      registerEventHandler: () => this.allowTransitionEnd_ = true,
+      deregisterEventHandler: () => this.allowTransitionEnd_ = false,
+      getOffsetLeft: () => this.tabElement_.current && this.tabElement_.current.offsetLeft,
+      getOffsetWidth: () => this.tabElement_.current && this.tabElement_.current.offsetWidth,
+      getContentOffsetLeft: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetLeft,
+      getContentOffsetWidth: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetWidth,
+      focus: () => this.tabElement_.current && this.tabElement_.current.focus(),
+
+      // activateIndicator, deactivateIndicator, and computeIndicatorClientRect
+      // are not needed in the adapter. TabIndicator keeps calls foundation.active
+      // and foundation.deactivate when this.props.activate updates.
+      // computeIndicatorClientRect seems redundant in mdc-tab
+    };
+  }
+
+  computeIndicatorClientRect = () => {
+    if (!this.tabIndicatorElement_.current) return;
+    return this.tabIndicatorElement_.current.computeContentClientRect();
+  }
+
+  computeDimensions = () => {
+    return this.foundation_.computeDimensions();
+  }
+
+  handleTransitionEnd = (evt) => {
+    this.props.onTransitionEnd(evt);
+
+    if (!this.allowTransitionEnd_) return;
+
+    this.foundation_.handleTransitionEnd_(evt);
+  }
+
+  render() {
+    const {
+      /* eslint-disable */
+      active,
+      previousActiveClientRect,
+      className,
+      fadeIndicator,
+      indicator,
+      minWidth,
+      onTransitionEnd,
+      stacked,
+      /* eslint-enable */
+      children,
+      ...otherProps
+    } = this.props;
+
+    const {
+      tabIndex,
+      ['aria-selected']: ariaSelected,
+    } = this.state;
+
+    return (
+      <button
+        className={this.classes}
+        role='tab'
+        aria-selected={ariaSelected}
+        tabIndex={tabIndex}
+        ref={this.tabElement_}
+        onTransitionEnd={this.handleTransitionEnd}
+        {...otherProps}
+      >
+        <span
+          className='mdc-tab__content'
+          ref={this.tabContentElement_}
+        >
+          {children}
+        </span>
+        {this.renderIndicator()}
+        <span className='mdc-tab__ripple'></span>
+      </button>
+    );
+  }
+
+  renderIndicator() {
+    const {
+      active,
+      fadeIndicator,
+      indicator,
+      previousActiveClientRect,
+    } = this.props;
+
+    if (indicator) {
+      const indicatorProps = {
+        active,
+        previousIndicatorClientRect: previousActiveClientRect,
+        ref: this.tabIndicatorElement_,
+      };
+      const Indicator = indicator(indicatorProps);
+      if (Indicator.type !== TabIndicator) {
+        new Error('this.props.indicator must be a function that returns an instance of TabIndicator');
```

Comment:
This will be documented in more detail in the readme

Reply:
HMMMMM